apis_core/generic/fields.py

```diff
@@ -0,0 +1,106 @@
+import re
+from datetime import date
+from typing import Callable, List, Tuple
+
+from django.db.models import CharField, DateField
+from django.forms import ValidationError
+
+from apis_core.history.models import APISHistoryTableBase
+from apis_core.utils import DateParser
+
+
+class GenericDateIntervalField(CharField):
+    def contribute_to_class(self, cls, name):
+        super().contribute_to_class(cls, name)
+        if not issubclass(cls, APISHistoryTableBase):
+            DateField(editable=False, blank=True, null=True).contribute_to_class(
+                cls, f"{name}_date_sort"
+            )
+            DateField(editable=False, blank=True, null=True).contribute_to_class(
+                cls, f"{name}_date_from"
+            )
+            DateField(editable=False, blank=True, null=True).contribute_to_class(
+                cls, f"{name}_date_to"
+            )
+
+
+class FuzzyDateParserField(GenericDateIntervalField):
+    def __init__(
+        self,
+        parser: Callable[[str], Tuple[date, date, date]] = DateParser.parse_date,
+        *args,
+        **kwargs,
+    ):
+        self.parser = parser
+        super().__init__(*args, **kwargs)
+
+    def pre_save(self, model_instance, add):
+        name = self.attname
+        value = getattr(model_instance, name)
+        if not getattr(model_instance, "skip_date_parsing", False):
+            try:
+                date, date_from, date_to = self.parser(value)
+                print(date)
+                setattr(model_instance, f"{name}_date_sort", date)
+                setattr(model_instance, f"{name}_date_from", date_from)
+                setattr(model_instance, f"{name}_date_to", date_to)
+            except Exception as e:
+                raise ValidationError(f"Error parsing date string: {e}")
+        return super().pre_save(model_instance, add)
+
+
+DEFAULT_DATE_REGEX = (r"(?P<day>\d{1,2})\.(?P<month>\d{1,2}).(?P<year>\d{1,4})",)
+
+
+class FuzzyDateRegexField(GenericDateIntervalField):
+    def __init__(
+        self,
+        regex_patterns: List[Tuple[str, str, str]] = [
```

Comment:
as discusssed remove the list here and add a doctring explaining the sequence of patterns: sort, from, to